com.unity.render-pipelines.universal/Editor/ShaderGraph/Targets/UniversalLitSubTarget.cs

```diff
@@ -247,302 +256,357 @@ public bool TryUpgradeFromMasterNode(IMasterNode1 masterNode, out Dictionary<Blo
                 { BlockFields.VertexDescription.Tangent, 11 },
                 { BlockFields.SurfaceDescription.BaseColor, 0 },
                 { normalBlock, 1 },
-                { specularMetallicBlock, specularMetallicId },
                 { BlockFields.SurfaceDescription.Emission, 4 },
                 { BlockFields.SurfaceDescription.Smoothness, 5 },
                 { BlockFields.SurfaceDescription.Occlusion, 6 },
                 { BlockFields.SurfaceDescription.Alpha, 7 },
                 { BlockFields.SurfaceDescription.AlphaClipThreshold, 8 },
             };
 
+            // PBRMasterNode adds/removes Metallic/Specular based on settings
+            if (m_WorkflowMode == WorkflowMode.Specular)
+                blockMap.Add(BlockFields.SurfaceDescription.Specular, 3);
+            else if (m_WorkflowMode == WorkflowMode.Metallic)
+                blockMap.Add(BlockFields.SurfaceDescription.Metallic, 2);
+
             return true;
         }
 
         #region SubShader
         static class SubShaders
         {
-            // Overloads to do inline PassDescriptor modifications
-            // NOTE: param order should match PassDescriptor field order for consistency
-            #region PassVariant
-            private static PassDescriptor PassVariant(in PassDescriptor source, PragmaCollection pragmas)
+            // SM 4.5, compute with dots instancing
+            public static SubShaderDescriptor LitComputeDotsSubShader(UniversalTarget target, WorkflowMode workflowMode, string renderType, string renderQueue, bool complexLit)
```

Comment:
Converted these static variables into static functions, as quite a bit of it is now dynamic.